corehq/apps/toggle_ui/models.py

```diff
@@ -1 +1,68 @@
-# This file is only here so that django will recognize that this is a valid app
+from django.db import models
+
+from corehq.toggles import NAMESPACE_USER, NAMESPACE_DOMAIN, NAMESPACE_OTHER
+
+
+class ToggleAuditManager(models.Manager):
+    def log_toggle_changes(self, slug, username, current_items, previous_items, randomness):
+        if current_items != previous_items:
+            added = current_items - previous_items
+            removed = previous_items - current_items
+            for action, namespaced_items in [(ToggleAudit.ACTION_ADD, added), (ToggleAudit.ACTION_REMOVE, removed)]:
+                self.log_toggle_action(slug, username, namespaced_items, action)
+
+        if randomness is not None:
+            self.create(
+                slug=slug, username=username, action=ToggleAudit.ACTION_UPDATE_RANDOMNESS,
+                randomness=randomness
+            )
+
+    def log_toggle_set(self, slug, username, namespaced_items):
```

Comment:
It looks like this function (and the one below) aren't actually used. Maybe consider YAGNI-ing them?

Reply:
https://github.com/dimagi/commcare-hq/pull/28815